environment/src/test/java/com/sequenceiq/environment/environment/flow/creation/handler/ResourceEncryptionInitializationHandlerTest.java

```diff
@@ -137,6 +157,40 @@ void acceptTestEnvironmentShouldNotBeUpdatedWhenEncryptionKeyUrlIsEmpty() {
         verifyEnvCreationEvent();
     }
 
+    @Test
+    void acceptTestEnvironmentShouldBeUpdatedWhenEncryptionKeyUrlIsPresent() {
+        doAnswer(i -> null).when(eventSender).sendEvent(baseNamedFlowEventCaptor.capture(), any(Headers.class));
+        Environment environment = new Environment();
+        environment.setParameters(newAzureParameters());
+        when(environmentService.findEnvironmentById(ENVIRONMENT_ID)).thenReturn(Optional.of(environment));
+        CreatedDiskEncryptionSet createdDiskEncryptionSet = new CreatedDiskEncryptionSet.Builder()
+                .withDiskEncryptionSetId("dummmy-set-id")
+                .build();
+        when(environmentEncryptionService.createEncryptionResources(any(), any())).thenReturn(createdDiskEncryptionSet);
+
+        underTest.accept(environmentDtoEvent);
+
+        verify(eventSender).sendEvent(baseNamedFlowEventCaptor.capture(), headersArgumentCaptor.capture());
+        verify(environmentService, times(1)).save(any());
```

Comment:
Instead of simply verifying the act of the `save()` invocation, please capture the method argument and then check of the DESid inside the capture `Environment` matches the one returned above.